### DISCUSSION TITLE
Is it possible to use h/w accelerator using 'nvv4l2h264enc' with kvssink?

### DISCUSSION THREAD

**Discussion by atg0831** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#970] [Q&A] [answered]
I want to use h/w accelerator encoding like 'nvv4l2h264enc' using kvssink, but does not work.
It works when to use x264enc instead of using 'nvv4l2h264enc'

** Logging **
```
[DEBUG] [21-02-2023 05:41:01:479.410 GMT] clientReadyHandler invoked
[TRACE] [21-02-2023 05:41:01:479.417 GMT] createKinesisVideoClientSync(): Awaiting for the Kinesis Video Client to become ready...
[TRACE] [21-02-2023 05:41:01:479.422 GMT] createKinesisVideoClientSync(): Kinesis Video Client is Ready.
[INFO ] [21-02-2023 05:41:01:479.431 GMT] try creating stream
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Fatal Python error: Aborted
```

Is it impossible to use 'nvv4l2h264enc' in kvssink yet?

**Comment by jdelapla** [comment, marked as answer]
kvssink has no limitation on what encoder you use with it. If the output is h264 and your pipeline has h264parse in it, any encoder can be used.

**Reply by atg0831** [reply]
Thank you for your reply.
But, not working yet.
I tried to find problem in my pipeline.
